### PR TITLE
DRV-104: Add reverse()

### PR DESCRIPTION
**Pull request by trevorsibanda**
- Adds fql reverse()

### CODE REVIEW

File: tests/test_query.py [hunk 1]
```diff
@@ -432,6 +432,32 @@ def test_merge(self):
     self.assertEqual(self._q(q3), {"a": 5, "b": 2, "c": 3})
     self.assertEqual(self._q(q4), {"a": 'a', "b": 'b', "c": 'c'})
 
+  def test_reverse(self):
+    #arrays
+    list = [0,1,2,3,4,5,6,7,8,9,10]
+    self.assertEqual(self._q(query.reverse(list)), list[::-1])
+    self.assertEqual(self._q(query.reverse(query.reverse(list))), list)
+    self.assertEqual(self._q(query.reverse(list[:0])), [])
```
Comment (evbruno) [inline tests/test_query.py:440]: I really love those python idiosyncrasies, but wouldn't be better the literal list `[10, 9, 8, ...]` here?  😊

File: tests/test_query.py [hunk 2]
```diff
@@ -567,7 +593,7 @@ def test_contains_str(self):
 
 
   def test_contains_str_regex(self):
-    self.assertEqual(self._q(query.contains_str_regex("faunadb", "f(\w+)a")), True)
+    self.assertEqual(self._q(query.contains_str_regex("faunadb", "f(\\w+)a")), True)
```
Comment (erickpintor) [inline tests/test_query.py:596]: This looks unrelated to the feature implemented.

File: tests/test_query.py (hunk 1 above)
Comment (erickpintor) [reply tests/test_query.py:440]: I also think literal lists are more readable here.
